tools/clang/lib/SPIRV/LowerTypeVisitor.cpp

```diff
@@ -176,6 +157,11 @@ bool LowerTypeVisitor::visitInstruction(SpirvInstruction *instr) {
     break;
   }
 
+  // Apply additional SPIR-V type transformations, including ensuring variables
+  // and function parameters have a pointer type.
+  SpirvTypeVisitor spirvTypeVisitor(context, opts);
+  spirvTypeVisitor.visitInstruction(instr);
```

Comment:
I am worried that readers cannot expect `bool LowerTypeVisitor::visitInstruction(SpirvInstruction *instr)` uses `SpirvTypeVisitor` object as a local variable.

I'd like to place this in `std::vector<uint32_t> SpirvBuilder::takeModule()` right after
```
LowerTypeVisitor lowerTypeVisitor(*astContext, context, spirvOptions);
mod->invokeVisitor(&lowerTypeVisitor);
// here
```

Is there any issue in that change?

Reply:
I am still concerned about this because we have an implicit rule to call the visitors only in the `takeModule()` method of SpirvBuilder. I know it can degrade the performance because it will visit all the instructions, but doing it in the `takeModule()` will be better for the long term maintenance.

I also took some time to understand this part. I believe later more readers would have the same issue.

I am sorry if I bother you but I'd like to update it
```
LowerTypeVisitor lowerTypeVisitor(*astContext, context, spirvOptions);
mod->invokeVisitor(&lowerTypeVisitor);
SpirvTypeVisitor spvTypeVisitor(context, spirvOptions);
mod->invokeVisitor(&spvTypeVisitor);
```

(if it does not have technical issue).

Reply:
As discussed offline, this change doesn't work in the current state, so I've moved it to a util function as requested.